lib/App.php

```diff
@@ -344,31 +346,121 @@ public function getConnection($required=false) {
         return $this->_connection;
     }
 
+	/**
+	 * ## Returns rendered contents of the view
+	 *
+	 * ### Definitions of localized views:**
+	 *
+	 * - default view: the original view path without localization, e.g 'main/index'
+	 * - localized view: the view path with locale code, e.g. 'main/en/index' or 'main/en-GB/index' whichever fits better.
+	 * - locale can be an ISO 639-1 language code ('en') optionally extended with a ISO 3166-1-a2 region ('en-GB')
+	 *
+	 * ### Rules for locale and language codes**
+	 *
+	 * - If current locale is 'en-GB', the path with 'en-GB' is preferred, otherwise 'en' is used. No other 'en-*' is used
+	 * - If current locale is 'en', the path with 'en' is used, no any 'en-*' is recognised.
+	 *
+	 * ### Locale selection
+	 *
+	 * - true: use current locale if the localized view exists, otherwise use the default view
+	 * - false: do not use localized view, even if exists. If the unlocalized view does not exist, an exception occurs.
+	 * - explicit locale: use the specified locale, as defined at 'true' case.
+	 *
+	 * Note: returns an error message rendered as a string on internal rendering errors or Exception
+	 *
+	 * @param string $viewName -- basename of a php view-file in the `views` directory, without extension and without localization code
+	 * @param array $params -- parameters to assign to variables used in the view
+	 * @param string $layout -- the layout applied to this render after the view rendered. If false, no layout will be applied.
```

Comment:
No, really "this (current) render". If you call App::render, ocassionally a tree of subviews/partial-views are rendered. But the subviews don't know anything about the initial top-level call. I can store the actual (explicit, computed or user-selected) locale here and use in all subviews. Of course this solution is not multithread-safe, but just now, I have no better idea. I don't want to instantiate a separate rendering context object, it would be way too complex for this simple framework..

Reply:
Applied to the result of the render...?

Reply:
The problem is that I'm not quite sure what _render_ refers to. But readers are likely to assume it is the function name, so the current documentation is probably fine.

Reply:
It's not okay that I couldn't describe it in a way that you could understand.